Tests/Tests/ArraySpec.swift

```diff
@@ -1,131 +1,83 @@
-                        let lhs: [String] = pure(append) <*> pure(string)
-                        let rhs: [String] = pure(append(string))
-
-                        return rhs == lhs
-                    }
-
-                    expect(property).to(hold())
-                }
-
-                // u <*> pure y = pure ($ y) <*> u
-                it("obeys the interchange law") {
-                    let property = generateString() { string in
-                        let lhs: [String] = pure(append) <*> pure(string)
-                        let rhs: [String] = pure({ $0(string) }) <*> pure(append)
-
-                        return lhs == rhs
-                    }
-
-                    expect(property).to(hold())
-                }
-
-                // u <*> (v <*> w) = pure (.) <*> u <*> v <*> w
-                it("obeys the composition law") {
-                    let property = generateArray() { array in
-                        let lhs = pure(append) <*> (pure(prepend) <*> array)
-                        let rhs = pure(curry(compose)) <*> pure(append)  <*> pure(prepend) <*> array
-
-                        return lhs == rhs
-                    }
-
-                    expect(property).to(hold())
-                }
-            }
-
-            describe("flatMap") {
-                // return x >>= f = f x
-                it("obeys the left identity law") {
-                    let property = generateString() { string in
-                        let lhs: [String] = pure(string) >>- compose(append, pure)
-                        let rhs: [String] = compose(append, pure)(string)
-
-                        return lhs == rhs
-                    }
-
-                    expect(property).to(hold())
-                }
-
-                // m >>= return = m
-                it("obeys the right identity law") {
-                    let property = generateArray() { array in
-                        let lhs = array >>- pure
-                        let rhs = array
-
-                        return lhs == rhs
-                    }
-
-                    expect(property).to(hold())
-                }
-
-                // (m >>= f) >>= g = m >>= (\x -> f x >>= g)
-                it("obeys the associativity law") {
-                    let property = generateArray() { array in
-                        let lhs = (array >>- compose(append, pure)) >>- compose(prepend, pure)
-                        let rhs = array >>- { x in compose(append, pure)(x) >>- compose(prepend, pure) }
+    func testApplicative() {
+        // pure id <*> v = v
+        property("identity law") <- forAll { (array: [String]) in
+            let lhs = pure(id) <*> array
+            let rhs = array
+
+            return lhs == rhs
+        }
+
+        // pure f <*> pure x = pure (f x)
+        property("homomorphism law") <- forAll { (string: String) in
+            let lhs: [String] = pure(append) <*> pure(string)
```

Comment:
Interestingly, generating functions _seems_ to be giving me a number of intermittent failures. Not sure what that's about.